rpcs3/Emu/RSX/rsx_utils.cpp

```diff
@@ -135,60 +167,60 @@ namespace rsx
 
 	double avconf::get_aspect_ratio() const
 	{
-		video_aspect v_aspect;
 		switch (aspect)
 		{
-		case CELL_VIDEO_OUT_ASPECT_16_9: v_aspect = video_aspect::_16_9; break;
-		case CELL_VIDEO_OUT_ASPECT_4_3: v_aspect = video_aspect::_4_3; break;
-		case CELL_VIDEO_OUT_ASPECT_AUTO:
-		default: v_aspect = g_cfg.video.aspect_ratio; break;
-		}
-
-		double aspect_ratio;
-		switch (v_aspect)
-		{
-		case video_aspect::_4_3: aspect_ratio = 4.0 / 3.0; break;
-		case video_aspect::_16_9: aspect_ratio = 16.0 / 9.0; break;
+		case CELL_VIDEO_OUT_ASPECT_16_9: return 16. / 9.;
+		case CELL_VIDEO_OUT_ASPECT_4_3: return 4. / 3.;
+		default: fmt::throw_exception("Invalid aspect ratio %d", aspect);
 		}
-		return aspect_ratio;
 	}
 
-	void avconf::upscale_to_aspect_ratio(int& width, int& height) const
+	size2u avconf::aspect_convert_dimensions(const size2u& image_dimensions) const
```

Comment:
why not just use a copy by value here and remove the result var?

Reply:
also, I feel like the old function name conveys what it actually does, while the new one is kinda cryptic

Reply:
How is scaling a 720p image to a 1080p or 4k size downscaling?
Accurate names would be something like aspect_ratio_correction_unconstrained and aspect_ratio_correction_constrained but that is too wordy. I'll think of something easier to read though.

Reply:
if you look closely you'll see that the function name was "upscale_to_aspect_ratio" xD

Reply:
I'd probably call it "expanded_to_aspect_ratio" or whatever

Reply:
like... Qt calls their size methods expandTo as well